Samples/PublicSamples/EchoBot/src/EchoBot/Bot/BotService.cs

```diff
@@ -114,35 +99,64 @@ public void Initialize()
                 name,
                 _settings.AadAppId,
                 _settings.AadAppSecret,
-                _graphLogger,
-                _logger);
+                _graphLogger);
+            //_logger);
+
+            var mediaPlatformSettings = new MediaPlatformSettings()
+            {
+                MediaPlatformInstanceSettings = new MediaPlatformInstanceSettings()
+                {
+                    CertificateThumbprint = _settings.CertificateThumbprint,
+                    InstanceInternalPort = _settings.MediaInternalPort,
+                    InstancePublicIPAddress = IPAddress.Any,
+                    InstancePublicPort = _settings.MediaInstanceExternalPort,
+                    ServiceFqdn = _settings.MediaDnsName
+                },
+                ApplicationId = _settings.AadAppId,
+                MediaPlatformLogger = _mediaPlatformLogger
+            };
+
+            var notificationUrl = new Uri($"https://{_settings.ServiceDnsName}:{_settings.BotInstanceExternalPort}/{HttpRouteConstants.CallSignalingRoutePrefix}/{HttpRouteConstants.OnNotificationRequestRoute}");
 
             builder.SetAuthenticationProvider(authProvider);
-            builder.SetNotificationUrl(_azureSettings.CallControlBaseUrl);
-            builder.SetMediaPlatformSettings(_azureSettings.MediaPlatformSettings);
+            builder.SetNotificationUrl(notificationUrl);
+            builder.SetMediaPlatformSettings(mediaPlatformSettings);
             builder.SetServiceBaseUrl(new Uri(AppConstants.PlaceCallEndpointUrl));
 
             this.Client = builder.Build();
             this.Client.Calls().OnIncoming += this.CallsOnIncoming;
             this.Client.Calls().OnUpdated += this.CallsOnUpdated;
         }
 
+        public async Task Shutdown()
+        {
+            _logger.LogWarning("Terminating all calls during shutdown event");
+            await this.Client.TerminateAsync();
+            this.Dispose();
+        }
+
         /// <summary>
         /// End a particular call.
         /// </summary>
         /// <param name="callLegId">The call leg id.</param>
         /// <returns>The <see cref="Task" />.</returns>
-        public async Task EndCallByCallLegIdAsync(string callLegId)
+        public async Task EndCallByCallLegIdAsync(string meetingId)
```

Comment:
This was just the name of the method and it wasn't hooked up in a controller. I added a delete method in the Calls controller and updated the variables to reflect threadId. Since you can only get call Id from the bot or graph, it is easier for the user to use threadId. They add the bot to the meeting with the joinUrl, which has the threadId in it. Now after joining, we return the threadId to make it easier for the user to use it in the delete call if they want to test that endpoint.